config/include.go

```diff
@@ -133,6 +225,174 @@ func (targetConfig *TerragruntConfig) Merge(sourceConfig *TerragruntConfig, terr
 	}
 }
 
+// DeepMerge performs a deep merge of the given sourceConfig into the targetConfig. Deep merge is defined as follows:
+// - For simple types, the source overrides the target.
+// - For lists, the two attribute lists are combined together in concatenation.
+// - For maps, the two maps are combined together recursively. That is, if the map keys overlap, then a deep merge is
+//   performed on the map value.
+// - Note that some structs are not deep mergeable due to an implementation detail. This will change in the future. The
+//   following structs have this limitation:
+//     - remote_state
+//     - generate
+// - Note that the following attributes are deliberately omitted from the merge operation, as they are handled
+//   differently in the parser:
+//     - dependency blocks (TerragruntDependencies) [These blocks need to retrieve outputs, so we need to merge during
+//       the parsing step, not after the full config is decoded]
+//     - locals [These blocks are not merged by design]
+func (targetConfig *TerragruntConfig) DeepMerge(sourceConfig *TerragruntConfig, terragruntOptions *options.TerragruntOptions) error {
+	// Merge simple attributes first
+	if sourceConfig.DownloadDir != "" {
+		targetConfig.DownloadDir = sourceConfig.DownloadDir
+	}
+
+	if sourceConfig.IamRole != "" {
+		targetConfig.IamRole = sourceConfig.IamRole
+	}
+
+	if sourceConfig.IamAssumeRoleDuration != nil {
+		targetConfig.IamAssumeRoleDuration = sourceConfig.IamAssumeRoleDuration
+	}
+
+	if sourceConfig.TerraformVersionConstraint != "" {
+		targetConfig.TerraformVersionConstraint = sourceConfig.TerraformVersionConstraint
+	}
+
+	if sourceConfig.TerraformBinary != "" {
+		targetConfig.TerraformBinary = sourceConfig.TerraformBinary
+	}
+
+	if sourceConfig.PreventDestroy != nil {
+		targetConfig.PreventDestroy = sourceConfig.PreventDestroy
+	}
+
+	if sourceConfig.RetryMaxAttempts != nil {
+		targetConfig.RetryMaxAttempts = sourceConfig.RetryMaxAttempts
+	}
+
+	if sourceConfig.RetrySleepIntervalSec != nil {
+		targetConfig.RetrySleepIntervalSec = sourceConfig.RetrySleepIntervalSec
+	}
+
+	if sourceConfig.TerragruntVersionConstraint != "" {
+		targetConfig.TerragruntVersionConstraint = sourceConfig.TerragruntVersionConstraint
+	}
+
+	// Skip has to be set specifically in each file that should be skipped
+	targetConfig.Skip = sourceConfig.Skip
+
+	// Handle list attributes by concatenatenation
+	if sourceConfig.Dependencies != nil {
+		resultModuleDependencies := &ModuleDependencies{}
+		if targetConfig.Dependencies != nil {
+			resultModuleDependencies.Paths = targetConfig.Dependencies.Paths
+		}
+		resultModuleDependencies.Paths = append(resultModuleDependencies.Paths, sourceConfig.Dependencies.Paths...)
+		targetConfig.Dependencies = resultModuleDependencies
+	}
+
+	if sourceConfig.RetryableErrors != nil {
+		targetConfig.RetryableErrors = append(targetConfig.RetryableErrors, sourceConfig.RetryableErrors...)
+	}
```

Comment:
Does this work OK if `targetConfig.RetryableErrors` is `nil`?

Reply:
Yes. Append handles `nil` correctly. See https://play.golang.org/p/5RRq6Owj_sS